src/js/components/DataTable/Body.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { TableRow, TableCell } from '../Table';
+import { InfiniteScroll } from '../InfiniteScroll';
+
+import Cell from './Cell';
+import { StyledDataTableBody, StyledDataTableRow } from './StyledDataTable';
+
+const Body = ({ columns, data, primaryProperty, size, theme, ...rest }) => (
+  <StyledDataTableBody size={size} theme={theme} {...rest}>
+    <InfiniteScroll
```

Comment:
This will be a problem with long grouped elements. I like your idea of adding a `loadMore` instead of automatically loading more results for that group.

Reply:
Agreed that this needs further work for long group expansions.